File: OnlineJudge/LeetCodeOJ/206 - Reverse Linked List/Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param {ListNode} head
    # @return {ListNode}
    def reverseList(self, head):
        if not head:
            return head
        
        curr = head
        prev = None
        next = None
        while curr:
            next = curr.next;
            curr.next = prev;
            prev = curr;
            curr = next;
            head = prev;
        
        return head

File: OnlineJudge/LeetCodeOJ/206 - Reverse Linked List/Reverse Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if (head == NULL)
            return NULL;
        ListNode *curr = head, *prev = NULL, *next = NULL;
        while (curr != NULL) {
            next = curr->next;
            curr->next = prev;
            prev = curr;
            curr = next;
            head = prev;
        }
        return head;
    }
};